### PR TITLE
cb_explore_adf: fixed bug when resetting cb_type + improvements

### PR DESCRIPTION
**Pull request by marco-rossi29**
-

### CODE REVIEW

File: vowpalwabbit/cb_explore_adf.cc [hunk 1]
```diff
@@ -754,26 +754,46 @@ base_learner* cb_explore_adf_setup(options_i& options, vw& all)
...
+  // Ensure serialization of cb_type in all cases.
   if (!options.was_supplied("cb_type"))
   {
     options.insert("cb_type", type_string);
     options.add_and_parse(new_options);
   }
 
   data->all = &all;
-  if (data->lambda < 0)  // Lambda should always be negative because we are using a cost basis.
+  if (data->lambda < 0)  // Lambda should always be postive because we are using a cost basis.
```
Comment (jackgerrits) [inline vowpalwabbit/cb_explore_adf.cc]:
```suggestion
  if (data->lambda < 0)  // Lambda should always be positive because we are using a cost basis.
```

Comment (marco-rossi29) [reply vowpalwabbit/cb_explore_adf.cc]: done, thanks!

**Comment by marco-rossi29** [comment]
@jackgerrits I added more changes. Take a look at the new commits. In particular:
Can you check if I did everything right in the header files in commit ed8b65a

Commit 431837c was discussed with @JohnLangford

File: vowpalwabbit/cb_adf.h [hunk 2]
```diff
@@ -1,9 +1,11 @@
 #pragma once
+#include <vector>
 
 LEARNER::base_learner* cb_adf_setup(VW::config::options_i& options, vw& all);
 
 namespace CB_ADF
 {
 CB::cb_class get_observed_cost(multi_ex& examples);
 void global_print_newline(vw& all);
+example* test_adf_sequence(std::vector<example*>& ec_seq);
```
Comment (jackgerrits) [inline vowpalwabbit/cb_adf.h]:
```suggestion
example* test_adf_sequence(multi_ex& ec_seq);
```

Comment (marco-rossi29) [reply vowpalwabbit/cb_adf.h]: done, thanks!

**Pull request merged**